### PR TITLE
Fix console launcher's `--uid` option

### LINKED ISSUE #4587: Fix console launcher `--uid` option
The discussion in https://github.com/junit-team/junit5/discussions/4583 combined with the linked issue https://github.com/eclipse-jdt/eclipse.jdt.core/issues/4053 shows that two bugs are hiding here. One is in JUnit:

When...
```
      --uid, --select-unique-id=UNIQUE-ID...
                             Select a unique id for test discovery. This option can be
                               repeated.
```
...is used as the sole selector, the console launcher shows a false error message reading:

```
Please specify an explicit selector option or use `--scan-class-path` or `--scan-modules`
```

There exists a work-around, though. Use `--select uid:UNIQUE-ID...` to prevent the error to happen.

According to what I read from https://github.com/eclipse-jdt/eclipse.jdt.core/issues/4053, the other bug seems to be in `ecj` and therefore cannot be addressed here.

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Prior to this change, unique id selectors weren't included in the list of explicit selectors - which resulted in a confusing error message:
```
Please specify an explicit selector option or use `--scan-class-path` or `--scan-modules`
```

Note that `--select uid:[...]` was not affected by this bug.

Fixes #4587

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)

### CODE REVIEW

File: platform-tests/src/test/java/org/junit/platform/console/tasks/DiscoveryRequestCreatorTests.java [hunk 1]
```diff
@@ -218,6 +221,21 @@ void convertsEngineOptions() {
...
+		options.setSelectedUniqueId(List.of(selectUniqueId("[engine:a]/[1:1]"), selectUniqueId("[engine:b]/[2:2]")));
+
+		var request = convert();
+		var uriSelectors = request.getSelectorsByType(UniqueIdSelector.class);
+
+		assertThat(uriSelectors) //
+				.extracting(UniqueIdSelector::getUniqueId) //
+				.containsExactly( //
+					UniqueId.forEngine("a").append("1", "1"), //
+					UniqueId.forEngine("b").append("2", "2") //
```
Comment (marcphilipp) [inline platform-tests/src/test/java/org/junit/platform/console/tasks/DiscoveryRequestCreatorTests.java:235]:
```suggestion
					UniqueId.parse("[engine:a]/[1:1]"), //, //
					UniqueId.parse("[engine:b]/[2:2]") //
```
